examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairAlreadyDiscoveredCommand.java

```diff
@@ -0,0 +1,30 @@
+package com.matter.controller.commands.pairing;
+
+import chip.devicecontroller.ChipDeviceController;
+import com.matter.controller.commands.common.CredentialsIssuer;
+
+public final class PairAlreadyDiscoveredCommand extends PairingCommand {
+  public PairAlreadyDiscoveredCommand(
+      ChipDeviceController controller, CredentialsIssuer credsIssue) {
+    super(
+        controller,
+        "already-discovered",
+        PairingModeType.ALREADY_DISCOVERED,
+        PairingNetworkType.ETHERNET,
+        credsIssue);
+  }
+
+  @Override
+  protected void runCommand() {
+    currentCommissioner()
+        .pairDeviceWithAddress(
+            getNodeId(),
+            getRemoteAddr().getHostAddress(),
+            getRemotePort(),
+            getDiscriminator(),
```

Comment:
I don't think we need to specify discriminator in pairDeviceWithAddress, but this need update all Java clients and Pairing on-network also translated to call this API finally.

Since we need to update a lot of componets if we remove discriminator from Java API pairDeviceWithAddress, I created https://github.com/project-chip/connectedhomeip/issues/24935 as the follow-up,

Reply:
Currently pairing on-network also use pairDeviceWithAddress in which local host address is used, this is wrong, that is why I am only  able to use 'pairing on-network' to pair a device on the same host, but not the device within the same network. I think we need to use DNS based pair to align with chip-tool implemenation.